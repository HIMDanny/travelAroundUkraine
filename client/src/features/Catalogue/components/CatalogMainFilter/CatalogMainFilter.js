import React from 'react';
import { styled, Button, Typography, Stack } from '@mui/material';
import Grid from '@mui/material/Unstable_Grid2';
import CloseIcon from '@mui/icons-material/Close';
import { CatalogFilterPrice, CatalogFilterTourists, CatalogFilterRegion } from '..';

const FilterWrapper = styled(Stack)(({ theme }) => ({
  margin: '0 auto',
  width: 252,
  padding: 0,

  [theme.breakpoints.up('tablet')]: {
    width: 540,
  },

  [theme.breakpoints.up('laptop')]: {
    width: 235,
  },
}));

const ShowButton = styled(Button)(() => ({
  padding: '12px 85px',
  alignSelf: 'center',
  marginTop: 40,
  maxWidth: 200,
}));

const ResetButton = styled((props) => (
  <Button
    variant="text"
    startIcon={
      <CloseIcon
        sx={{
          width: '14px',
          height: '14px',
          color: 'gray',
        }}
      />
    }
    {...props}
  />
))(({ theme }) => ({
  padding: '5px 15px',
  marginTop: '12px',
  border: 'none',
  background: 'none',
  fontWeight: 500,
  fontSize: 12,
  textTransform: 'none',
  color: theme.palette.text.primary,
  alignSelf: 'center',

  '&:hover': {
    background: 'none',
    filter: 'brightness(1.5)',
    color: theme.palette.text.primary,
  },
}));

const CatalogMainFilter = () => (
  <FilterWrapper>
    <Typography variant="h3">Filter</Typography>

    <Grid container columnSpacing={5} sx={{ p: 0 }}>
      <Grid spacing={4} item xs={12} tablet={6} laptop={12}>
        <CatalogFilterPrice />
        <CatalogFilterTourists />
      </Grid>

      <Grid item xs={12} tablet={6} laptop={12}>
        <CatalogFilterRegion />
      </Grid>
    </Grid>

    <ShowButton>Filter</ShowButton>
    <ResetButton>Reset filter</ResetButton>
  </FilterWrapper>
);

export default CatalogMainFilter;
