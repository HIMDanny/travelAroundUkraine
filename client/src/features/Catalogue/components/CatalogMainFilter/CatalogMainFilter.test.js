import { render } from '@testing-library/react';
import CatalogMainFilter from './CatalogMainFilter';

jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Stack', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <h3>{children}</h3>);
jest.mock('@mui/material/Unstable_Grid2', () => ({ children }) => <div>{children}</div>);

jest.mock('../CatalogFilterPrice/CatalogFilterPrice', () => () => <div>CATALOGUE FILTER PRICE</div>);
jest.mock('../CatalogFilterTourists/CatalogFilterTourists', () => () => <div>CATALOGUE FILTER TOURISTS</div>);
jest.mock('../CatalogFilterRegion/CatalogFilterRegion', () => () => <div>CATALOGUE FILTER REGION</div>);

describe('CatalogMainFilter snapshot test', () => {
  test('should CatalogTourCard match snapshot', () => {
    const { asFragment } = render(<CatalogMainFilter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
