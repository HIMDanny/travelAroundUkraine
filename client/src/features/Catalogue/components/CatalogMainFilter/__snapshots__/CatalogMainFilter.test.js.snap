// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogMainFilter snapshot test should CatalogTourCard match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Filter
    </h3>
    <div>
      <div>
        <div>
          CATALOGUE FILTER PRICE
        </div>
        <div>
          CATALOGUE FILTER TOURISTS
        </div>
      </div>
      <div>
        <div>
          CATALOGUE FILTER REGION
        </div>
      </div>
    </div>
    <button>
      Filter
    </button>
    <button>
      Reset filter
    </button>
  </div>
</DocumentFragment>
`;
